sections/06-vibrant-bodies.md

```diff
@@ -0,0 +1,201 @@
+
+Programs made with Flocking and Aconite are data, not just code. This means, in particular, that they can be "understood" by programs and modified algorithmically as data, using the strategies that are typically used for other kinds of data processing. One particular technique is used extensively in Flocking programs to make them more amenable to material transformation: merging. In Flocking, instruments and signal processors are defined as "trees" of data that represent the connections amongst individual "unit generators," or signal producing/modulating algorithms. Even the compositional or "meta-level" material of a piece can be defined as a tree of signals, as in _Font Màgica<sup>10</sup>. By structuring a program as nested collections of named values, simple algorithms can be created (including the ones that are built directly into Flocking and its underlying support infrastructure) to "graft" or merge different signal trees together. A given instance of a signal tree created by one artist can be reconfigured without "forking" or changing the original source<sup>11</sup>, nor requiring there even to be an intent or active design to support such adaptation. Where free software legally mandates sharing, code-as-data participates in a form of implicit, integral gift economy where each work holds the potential to serve as the material for new ones.
+
+Like Amy Twigger-Holroyd's spectrum of treatments for knitted objects<sup>12</sup>, merging and other data processing strategies in Flocking provide a means to materially transform the nature and logic of a program, allowing its capacities and energies to be changed, re-channeled, or vectorized into new artistic forms. Notably, this "graftability" of my software can increasingly be applied at any point in the creative life of a program, from its inception as source code right up to the runtime manifestation of a program<sup>13</sup>.
+
+    {
```

Comment:
Comment from David: "Is it worth commenting the code a bit? I see the figure explanation, but it might help to have each line commented out."